src/lib/comms/include/sol-netctl.h

```diff
@@ -209,6 +216,60 @@ enum sol_netctl_state {
 };
 
 /**
+ * @brief agent input struct
+ *
+ * This struct contains the information of agent input.
+ */
+typedef struct sol_netctl_agent_input {
+    /**
+     * @brief The agent prompt type
+     */
+    char *type;
+    /**
+     * @brief The agent input value
+     */
+    char *input;
+} sol_netctl_agent_input;
+
+/**
+ * @brief agent callback functions
+ *
+ * This struct contains the callback functions of agent.
+ */
+typedef struct sol_netctl_agent {
+    /**
+     * @brief connection error callback used to inform connection failure
+     *
+     * @param data the user data
+     * @param service the connection failure service
+     * @param error the error information
+     */
+    void (*report_error)(void *data, const struct sol_netctl_service *service,
+        const char *error);
+    /**
+     * @brief connection input callback used to inform connection login input
+     *
+     * @param data the user data
+     * @param service the connection login input service
+     * @param inputs the vector of login input type
+     */
+    void (*request_input)(void *data, const struct sol_netctl_service *service,
+        const struct sol_vector *inputs);
```

Comment:
seing the sample this method is not clear to me. This method is called when connman needs info for a specific input right ? In this implementation the user (that have registered the agent) will be called with a list of inputs, then they will need to create another list of inputs with the answers and then call `sol_netctl_request_input` ? Why we simply don't allow the user fill the answer and we use it ?

What I told is right ?